common/metrics/config.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import "fmt"
+
+type HistogramMigration struct {
+	Default HistogramMigrationMode            `yaml:"default"`
+	Keys    map[string]HistogramMigrationMode `yaml:"keys"`
+}
+
+func (h HistogramMigration) EmitTimer(key string) bool {
+	mode, ok := h.Keys[key]
+	if ok {
+		return mode.EmitTimer()
+	}
+	return h.Default.EmitTimer()
+}
+func (h HistogramMigration) EmitHistogram(key string) bool {
+	mode, ok := h.Keys[key]
+	if ok {
+		return mode.EmitHistogram()
+	}
+	return h.Default.EmitHistogram()
+}
+
+// HistogramMigrationMode is a pseudo-enum to provide unmarshalling config and helper methods.
+// It should only be created by YAML unmarshaling, or by getting it from the HistogramMigration map.
+// Zero values from the map are valid, they are just the default mode (not the configured default).
+//
+// By default / when not specified / when an empty string, it currently means "both",
+// but this will change when timers are fully deprecated and removed.
+type HistogramMigrationMode string
+
+func (h *HistogramMigrationMode) UnmarshalYAML(read func(any) error) error {
+	var value string
+	if err := read(&value); err != nil {
+		return fmt.Errorf("cannot read histogram migration mode as a string: %w", err)
+	}
+	switch value {
+	case "timer", "histogram", "both":
```

Comment:
surprisingly, as tests show, empty values don't call this unmarshal func at all.  so they don't "need" to be handled.

that seems rather Bad™, but I don't think it'll change while we're on yaml/v2.  and tests should catch if/when it changes in later major versions / different libraries.